R/listWebData.R

```diff
@@ -0,0 +1,107 @@
+#' @param datasetName A character string. Represent the dataset of interest for download. datasetName allow to derived
+#'        url and password from the urltble.
+#' @rd
+#' @return List available data from a specific dataset
+#'
+#' @importFrom RCurl getURL url.exists
+#' @importFrom data.table setkey
+#' @export
+#' @docType methods
+#'
+#' @author Melina Houle
+#' @examples
+#'
+#' dt<- data.table(dataset = c("NFDB"),
+#'                url = c("http://cwfis.cfs.nrcan.gc.ca/downloads/nfdb/fire_poly/current_version/"),
+#'                password= c(NA))
+#'
+#' setwd(tempdir())
+#' listWebData(dt, datasetName = "NFDB")
+listWebData<-function(urltble,datasetName){
+
+  if(missing(datasetName)) stop("You must provide dataset name to derive url of interest")
+
+  setkey(urltble, "dataset")
+  password<- as.character(urltble[.(datasetName)][, 'password', with=FALSE])
+  path2data<- as.character(urltble[.(datasetName)][, 'url', with=FALSE])
+
+    if (!is.na(password)) {path2data <- paste(unlist(strsplit(path2data, "//"))[1],"//", password, "@",
+                                            unlist(strsplit(path2data, "//"))[2], sep="")
+
+                        if (!url.exists(as.character(path2data))) stop("url and/or password is erroneous")
+
+                        file.list <- getURL(path2data, userpwd = password, ftp.use.epsv = FALSE, dirlistonly = TRUE)
+                        file.list <- strsplit(file.list, "\r\n")[[1]]
+  } else {
+
+    if (!url.exists(as.character(path2data))) stop("url is erroneous")
+
+    typeConn <- paste(unlist(strsplit(path2data, "//"))[1],"//", sep ="")
+    if (typeConn == "http://") {file.list <- readHTMLTable(path2data, skip.rows=1:2)[[1]]$Name
+                                file.list<- as.character(file.list)
+    }
+    if (typeConn == "ftp://")  {file.list <- getURL(path2data, userpwd = NULL,
+                                                   ftp.use.epsv = FALSE, dirlistonly = TRUE)
+                                 file.list <- strsplit(file.list, "\r\n")[[1]]
+    }
+  }
+  return(file.list)
+}
+
+##################################################################################
+#' Table of dataset accessible using url
+#'
+#' An R object that stores dataset with their respective url and username/password. This table makes the link between
+#' dataset name, url and username/password. By only poviding the dataset name, listWebdata and dwdData functions extract url
+#' username/password to access, extract and download data.
+#'
+#' @rdname webdataset
+#' @rd
+#' @return Object containing dataset available for download.
+#'
+#' @importFrom data.table data.table
+#' @export
+#' @docType methods
+#'
+#' @author Melina Houle
+#' @examples
+#' dt<- data.table(dataset = c("NFDB"),
+#'                url = "http://cwfis.cfs.nrcan.gc.ca/downloads/nfdb/fire_poly/current_version/",
+#'                password= NA,
+#'                files = "NFDB_poly.zip")
+urls<- data.table(dataset = c("AHCCD_daily",
```

Comment:
why is this object being created in the package, especially with such a common name? it should either be a function that returns the data.table, or if it really needs to be pre-computed the object can be stored in the package environment and queried with a function.